### DISCUSSION TITLE
Using semi-hard negative miner with NT-Xent loss

### DISCUSSION THREAD

**Discussion by lemuelatQSA** [KevinMusgrave/pytorch-metric-learning#559] [Q&A] [answered]
I was analyzing what happens when we use `BatchEasyHardMiner(neg_strategy=BatchEasyHardMiner.SEMIHARD)` along with the `NTXentLoss`, and I realized that it basically computes

$$
  \mathcal{L_{NT}} = - \log \frac
  {\exp( sim(x_a, x_p) / \tau )}
  {\exp( sim(x_a, x_p) / \tau ) + \exp( sim(x_a, x_n) / \tau )}
$$

Where $sim(\cdot)$ represents the cosine similarity, and $x_a,x_p,x_n$ are the anchor, positive and negative respectively. Is there anyone who can confirm that? Can we still call this "NT-Xent loss"? I believe so, since it looks like a simplified version of it. Indeed, not involving other negatives from other classes will slower the convergence since we're optimizing the distance only w.r.t. $x_n$ class.

Any comment/opinion is appreciated! :)

**Comment by KevinMusgrave** [comment, marked as answer]
That looks correct to me. The [easy/hard mining paper](https://openaccess.thecvf.com/content_WACV_2020/papers/Xuan_Improved_Embeddings_with_Easy_Positive_Triplet_Mining_WACV_2020_paper.pdf) has a relevant section:

![image](https://user-images.githubusercontent.com/2314891/210333743-5f7d59ac-854b-4bda-8944-c24fb3355e6c.png)

I think it's fine to call it "NT-Xent" or "InfoNCE" because that's what you would call it if you did no mining and the batch naturally had only 1 negative.

**Reply by lemuelatQSA** [reply]
I agree. Thank you!
